test/csuite/wt12015_backup_corruption/main.c

```diff
@@ -0,0 +1,442 @@
+        do_work_before_failure(conn, session);
+
+        /* Die before finishing the next checkpoint. */
+        printf("Setting the failpoint...\n");
+        testutil_check(
+          session->reconfigure(session, "debug=(checkpoint_fail_before_turtle_update=true)"));
+        testutil_sentinel(NULL, EXPECT_ABORT);
+        testutil_check(session->checkpoint(session, NULL));
+        testutil_remove(EXPECT_ABORT);
+
+        /* We should die before we get here. */
+        testutil_die(ENOTRECOVERABLE, "The child process was supposed be dead by now!");
+    }
+
+    /* Parent. */
+
+    /* Wait for the child to die. */
+    testutil_assert(waitpid(pid, &status, 0) > 0);
+    printf("-- crash --\n");
+
+    /* Save the database directory. */
+    testutil_copy(WT_HOME_DIR, "save");
+
+    /* Do more work, create another backup, and verify it. */
+    do_work_after_failure(true);
+}
+
+/*
+ * run_test2 --
+ *     Run a test with force stop.
+ */
+static void
+run_test2(void)
+{
+    WT_CONNECTION *conn;
+    WT_SESSION *session;
+    pid_t pid;
+    int status;
+
+    printf("\n%s: Test crashing during force-stop after incremental backup\n", __func__);
+
+    testutil_remove(EXPECT_ABORT);
+    testutil_assert_errno((pid = fork()) >= 0);
+
+    if (pid == 0) { /* Child. */
+        testutil_recreate_dir(WT_HOME_DIR);
+        testutil_wiredtiger_open(opts, WT_HOME_DIR, env_config, NULL, &conn, false, false);
+        testutil_check(conn->open_session(conn, NULL, NULL, &session));
+        testutil_check(session->create(session, TABLE_URI, TABLE_CONFIG));
+
+        /* Do some work, while creating checkpoints and doing backups. */
+        do_work_before_failure(conn, session);
+
+        /* Die before finishing the next checkpoint. */
+        printf("Setting the failpoint...\n");
+        testutil_check(
+          session->reconfigure(session, "debug=(checkpoint_fail_before_turtle_update=true)"));
+        testutil_sentinel(NULL, EXPECT_ABORT);
+        testutil_backup_force_stop(session);
+        testutil_remove(EXPECT_ABORT);
+
+        /* We should die before we get here. */
+        testutil_die(ENOTRECOVERABLE, "The child process was supposed be dead by now!");
+    }
+
+    /* Parent. */
+
+    /* Wait for the child to die. */
+    testutil_assert(waitpid(pid, &status, 0) > 0);
+    printf("-- crash --\n");
+
+    /* Save the database directory. */
+    testutil_copy(WT_HOME_DIR, "save");
+
+    /* Do more work, create another backup, and verify it. */
+    do_work_after_failure(false);
+}
+
+/*
+ * usage --
+ *     Print usage help for the program.
+ */
+static void
+usage(void)
+{
+    fprintf(stderr, "usage: %s%s [-n] [-s SCENARIO_NUM]\n\n", progname, opts->usage);
+    fprintf(stderr, "Test-specific options:\n");
+    fprintf(stderr, "    -n    disable log\n");
+    fprintf(stderr, "    -s N  set the scenario number (use 0 for all scenarios)\n");
+    exit(EXIT_FAILURE);
+}
+
+/*
+ * main --
+ *     The entry point for the test.
```

Comment:
It would be helpful if there was a longer comment somewhere, maybe here, that explained a bit about what this program is testing in more detail.

Reply:
Done. I added a short description here, which captures the goal of the test at a high level.